zwave_js_server/model/value.py

```diff
@@ -47,8 +47,12 @@ def get_value_id(node: "Node", event_data: ValueDataType) -> str:
     command_class = event_data["commandClass"]
     endpoint = event_data.get("endpoint") or "00"
     property_ = event_data["property"]
-    property_key_name = event_data.get("propertyKeyName") or "00"
-    return f"{node.node_id}-{command_class}-{endpoint}-{property_}-{property_key_name}"
+    # Some values that have a property key don't have a property key name, so we will
+    # fall back to property key if it exists
+    property_key = (
```

Comment:
I'm thinking it's best to be consistent and always include both property key and property key name. Is there a downside to that? Then we know what each position represents.

Reply:
The main downside is that all places where we call `get_value_id` have to be updated to include both, otherwise no

Reply:
Do you mean if the return value is parsed?

Reply:
sorry what I meant was that we would have to update `entity.get_zwave_value` to accept the new parameter and make the corresponding change everywhere. We'd also have to know both the property key and the key name for all values.

Reply:
Or we update our approach for value IDs so that we can search for a value based on partial information

Reply:
I think `get_value_id` should not accept an (event) dictionary but named arguments.

Reply:
Whether it comes in via an event dictionary or as named parameters, if we want to include both parameters in the id, anywhere where we need to get a value id we would have to supply both

Reply:
What if instead of calling `get_value_id` directly in Home Assistant, we had a method that allowed a subset of parameters to be provided and looped through the Values until it found a matching one, returning the corresponding `value_id`? That would solve the problem of putting all of the parameters in the value_id and not requiring HA to know every parameter in order to get the Value

Reply:
`propertyKey` is optional in Z-Wave JS just as `propertyKeyName` so the former would only be required when it's present and matters. Currently we supply `propertyKeyName` in 8 places in Home Assistant out of 30 calls to `get_zwave_value`.

The question about who should know what properties are needed is separate from changing this function. I'm not sure about abstracting that more. If you know what value you want you should know how to identify it, I think.

Reply:
I see, so you are suggesting we:
1) Change get_value_id to accepted named arguments instead of a dict
2) Add propertyKey and propertyKeyName to the value ID string
2) Change those 8 places to pass both propertyKey and propertyKeyName

Is that correct?

Reply:
Yes, that's basically my suggestion. 👍